uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-balance.report.component.spec.js

```diff
@@ -0,0 +1,213 @@
+        beforeEach(function () {
+          setupController();
+          spyOn(ctrl, 'loadReportPage').and.callThrough();
+          ctrl.$onInit();
+          $rootScope.$digest();
+        });
+
+        // .loadReportPage() is detailed below.
+        it('request page 1 of the report using .loadReportPage()', function () {
+          expect(ctrl.loadReportPage).toHaveBeenCalledWith(1);
+        });
+      });
+    });
+
+    describe('.loadReportPage()', function () {
+      beforeEach(function () {
+        setupController();
+      });
+
+      describe('when loading a report`s page', function () {
+        var page = 10;
+
+        beforeEach(function () {
+          setupControllerAndRunOnInit();
+          ctrl.loadReportPage(page);
+        });
+
+        it('sets .loading.report to true', function () {
+          expect(ctrl.loading.report).toBe(true);
+        });
+
+        it('calls leaveBalanceReport.all with the page parameters', function () {
+          expect(leaveBalanceReport.all).toHaveBeenCalledWith(
+            { managed_by: loggedInContactId },
+            { page: page, size: defaultReportSize }
+          );
+        });
+
+        describe('when finishing loading a report`s page', function () {
+          var expected, count;
+
+          beforeEach(function () {
+            var absenceTypes = absenceTypeMock.all().values;
+            var balanceReport = reportMockData.all().values;
+            count = reportMockData.all().count;
+
+            // reversing absence_types to see if component sorts them in the
+            // proper order.
+            balanceReport.forEach(function (r) {
+              r.absence_types.reverse();
+            });
+
+            // sets the balance report in an expected manner.
+            // each record`s .absence_types should be sorted in the same order
+            // as the component`s .absenceTypes array.
+            expected = balanceReport.map(function (record) {
+              record = Object.assign({}, record);
+
+              record.absence_types = absenceTypes.map(function (sortedType) {
+                return record.absence_types.find(function (unsortedType) {
+                  return unsortedType && parseInt(unsortedType.id) === parseInt(sortedType.id);
+                });
+              });
+
+              return record;
+            });
+
+            ctrl.loadReportPage(page);
+            $rootScope.$digest();
+          });
+
+          it('sets .loading.report to false', function () {
+            expect(ctrl.loading.report).toBe(false);
+          });
+
+          it('sets .reportCount to the total number of records', function () {
+            expect(ctrl.reportCount).toBe(count);
+          });
+
+          it('sets .report to the response of AbsenceType.all call', function () {
+            expect(ctrl.report.length).toEqual(expected.length);
+          });
+
+          it('sorts .report[].absence_types in the same order as .absenceTypes', function () {
```

Comment:
just to make it clear, you should not reference any implementation details in the title tests, unless a test is really specifically about some implementation detail (like if something is an array, or an object, say). Use english-like sentences